packages/frameworks-astro/src/config.ts

```diff
@@ -0,0 +1,3 @@
+import type { AuthConfig } from "@auth/core/types"
+
+export const defineConfig = (config: AuthConfig) => config
```

Comment:
Thoughts on adding support for a function overload to additionally allow passing in a handler that exposes the request context & returns an auth config? Or alternatively add it to `AstroAuth` with our own custom api endpoints.

The primary reason being (That I have run into myself) when using [Cloudflare D1](https://developers.cloudflare.com/d1/) you only have access to the database binding at runtime. Without this you will need to disable api endpoint injection & add it that way somehow?

## Example

```ts
// auth.config.ts
import { defineConfig } from '@auth/astro/config';
import { D1Adapter } from "@auth/d1-adapter"

export default defineConfig((ctx) => {
	return {
		adapter: D1Adapter(ctx.locals.runtime.env.YOUR_DATABASE_BINDING),
		providers: [],
	};
})
```

Reply:
That doesn't sounds impossible! I'd definitely love to have good cf workers support. I'll see what I can come up with quickly

Reply:
I may rework this a little more, but I'm pretty happy with what we have now. You can now pass a function accepting `APIContext` and return your auth config on-demand. I also need to add docs for this override